src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseApprovalService.cs

```diff
@@ -113,92 +113,155 @@ IEmailTemplateService emailTemplateService
             Guid releaseId,
             ReleaseStatusCreateRequest request)
         {
-            return await _persistenceHelper
-                .CheckEntityExists<Release>(releaseId, ReleaseChecklistService.HydrateReleaseForChecklist)
+            return await _context
+                .Releases
+                .HydrateReleaseForChecklist()
```

Comment:
It seems a bit strange what we do this here, when I would have thought the `ReleaseCheckListService` it should be responsible for checking the release is hydrated. I think I would change things to not pass the release around in the first place, and pass the unique identifier of the release instead.

Probably not one for this PR though.

I found the extension method for `HydrateReleaseForChecklist` a little strange in the way that its scope now makes this method accessible to anywhere in the Admin, seen when checking available methods of `_context.Releases`.

Reply:
Yup that's interesting.  I'll leave it for now, but it's definitely a sign I guess that we're maybe not using the ReleaseChecklistService in the best way.